utilities/src/parallel.rs

```diff
@@ -17,9 +17,9 @@
 use crate::{boxed::Box, vec::Vec};
 
 pub struct ExecutionPool<'a, T> {
-    #[cfg(feature = "parallel")]
+    #[cfg(not(feature = "serial"))]
     jobs: Vec<Box<dyn 'a + FnOnce() -> T + Send>>,
-    #[cfg(not(feature = "parallel"))]
+    #[cfg(feature = "serial")]
```

Comment:
If we're making `add_job` take in `Send` closures, then this can be deduplicated:

```suggestion
    jobs: Vec<Box<dyn 'a + FnOnce() -> T + Send>>,
```